external/fv3fit/fv3fit/sklearn/transformer.py

```diff
@@ -0,0 +1,95 @@
+import joblib
+import numpy as np
+import os
+from sklearn.preprocessing import StandardScaler
+from sklearn.base import TransformerMixin
+from typing import Sequence
+import yaml
+
+from fv3fit._shared.predictor import Reloadable
+from fv3fit._shared import (
+    get_dir,
+    put_dir,
+)
+from fv3fit._shared import io
+
+
+def _ensure_sample_dim(x: np.ndarray) -> np.ndarray:
+    if x.ndim == 1:
+        return x.reshape(1, -1)
+    else:
+        return x
+
+
+@io.register("sk-transformer")
+class SkTransformer(Reloadable):
+    """ Used to encode higher-dimension inputs into a
+    lower dimension latent space and decode latent vectors
+    back to the original feature space.
+
+    """
+
+    _TRANSFORMER_NAME = "sk_transformer.pkl"
+    _SCALER_NAME = "sk_scaler.pkl"
+    _METADATA_NAME = "metadata.yaml"
+
+    def __init__(
+        self,
+        transformer: TransformerMixin,
+        scaler: StandardScaler,
+        enforce_positive_outputs: bool = False,
+    ):
+        self.transformer = transformer
+        self.scaler = scaler
+        self.enforce_positive_outputs = enforce_positive_outputs
+
+    def predict(self, x: Sequence[np.ndarray]) -> Sequence[np.ndarray]:
+        original_feature_sizes = [feature.shape[-1] for feature in x]
+        x_concat = np.concatenate(x, axis=-1)
+        encoded = self.encode(x_concat)
+        decoded = self.decode(encoded)
+        decoded_split_features = np.split(
+            decoded, np.cumsum(original_feature_sizes[:-1]), axis=-1
+        )
+
+        return decoded_split_features
+
+    def encode(self, x):
+        x = _ensure_sample_dim(x)
```

Comment:
Is this the right level to enforce the need for a sample dimension?  If, so I'd add some tests related to the `SkTransformer` and documentation that it expects a certain shape of data.

Reply:
Yes, the sklearn models seem to require this while IIRC keras models are smarter about figuring it out if you only give a single sample. I'll make this clearer and add a test.

Reply:
To clarify, the behavior here is that it can either take inputs with or without a sample dim so that the behavior is similar to the base keras models (LMK if I am wrong about that).

Reply:
I don't remember off the top of my head, but thanks for adding some documentation on tests around the expectation!